### ISSUE TITLE
Service account: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by flozzone** [terraform-google-modules/terraform-google-kubernetes-engine#2141]
### TL;DR

Sometimes creating GCP service accounts can fail with Terraform due to GCPs eventual consistency model (See [terraform-provider-google#18087](https://github.com/hashicorp/terraform-provider-google/issues/18087)). 
After running apply again, it fails because the service account already exists. 

A possible workaround has been added to [Terraform/google_service_account](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_service_account#create_ignore_already_exists) module to ignore an already existing service account by passing `create_ignore_already_exists=true` when creating the `google_service_account`

### Expected behavior

Service account gets created or at least it would take an already existing one if running apply again

### Observed behavior

Terraform complains about an already existing GCP service account

### Terraform Configuration

```hcl
module "app_workload_identity" {
  source              = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
  name                = "some-sa-name"
  namespace           = "some-ns"
  project_id          = "some-project"
  roles               = []
}
```

### Terraform Version

```sh
Terraform v1.8.4
on linux_amd64
```
